packages/palette/src/utils/useCursor.ts

```diff
@@ -0,0 +1,37 @@
+import { useMemo, useState } from "react"
+
+const mapCursorToMax = (cursor: number, max: number) => {
+  if (max === 0) return 0
+  if (cursor >= 0) return cursor % max
+  const modulo = cursor % max
+  if (modulo === 0) return Math.abs(modulo) // -0
+  return modulo + max
+}
+
+/**
+ * useCursor
+ * Cycles an index within some max length
```

Comment:
Don't you have something like this extracted into a lib? Might be better to use that unless you can imagine this hook being used often elsewhere in palette code. If so, might be good to add a light description to this comment with a real world application for those unfamilar with `cursor` as a concept.

Reply:
Yeah — [I just inlined it since it's just simple/small](https://github.com/dzucconi/use-cursor). You're probably right.

I find I lean on this pattern for handling literally anything in a UI with next/previous since it's one of those things that is easy to screw up or forget, so I can for instance see myself using it in the future.